[CS2113-W10-4] uNivUSaver

DanLinhHuynh-Niwashi opened the pull request:
uNivUSaver is a CLI-based software that helps students to develop a better habit of managing money.

dZhei98 commented on src/main/java/seedu/Main.java:
```diff
@@ -0,0 +1,76 @@
+/*
```
I would recommend removing this file since it is entirely commented out.

dZhei98 commented on src/main/java/seedu/command/AddExpenseCommand.java:
```diff
@@ -0,0 +1,107 @@
+    }
+
+    @Override
+    public List<String> execute() {
```
This function seems to have multiple functionalities embedded. You can consider refactoring this.

DanLinhHuynh-Niwashi replied:
Created a new issue. Thank you!

dZhei98 commented on src/main/java/seedu/command/DeadAddTranctionCommand.java:
```diff
@@ -0,0 +1,51 @@
+/*
```
Similarly, this entire file has been commented out. You can consider removing this,

dZhei98 commented on src/main/java/seedu/main/Main.java:
```diff
@@ -0,0 +1,227 @@
+        categories = new CategoryList();
+        transactions = new TransactionList();
+
+        HelpCommand helpCommand = new HelpCommand();
+        parser.registerCommands(helpCommand);
+
+        AddCategoryCommand addCategoryCommand = new AddCategoryCommand(categories);
+        parser.registerCommands(addCategoryCommand);
+
+        parser.registerCommands(new AddIncomeCommand(transactions));
+
+        parser.registerCommands(new AddExpenseCommand(transactions));
+
+
+        ViewCategoryCommand viewCategoryCommand = new ViewCategoryCommand(categories);
+        parser.registerCommands(viewCategoryCommand);
+
+        DeleteCategoryCommand deleteCategoryCommand = new DeleteCategoryCommand(categories);
+        parser.registerCommands(deleteCategoryCommand);
+
+        DeleteTransactionCommand deleteTransactionCommand = new DeleteTransactionCommand(transactions);
+        parser.registerCommands(deleteTransactionCommand);
+
+        ViewExpenseCommand viewExpenseCommand = new ViewExpenseCommand(transactions);
+        parser.registerCommands(viewExpenseCommand);
+
+        ViewIncomeCommand viewIncomeCommand = new ViewIncomeCommand(transactions);
+        parser.registerCommands(viewIncomeCommand);
+
+        ViewTotalCommand viewTotalCommand = new ViewTotalCommand(transactions);
+        parser.registerCommands(viewTotalCommand);
+
+        HistoryCommand historyCommand = new HistoryCommand(transactions);
+        parser.registerCommands(historyCommand);
+
+        ByeCommand byeCommand = new ByeCommand();
+        parser.registerCommands(byeCommand);
```
You may want to consider refactoring this to a new file/function

DanLinhHuynh-Niwashi replied:
Created a new issue. Thank you!

dZhei98 commented on src/main/java/seedu/main/Parser.java:
```diff
@@ -0,0 +1,165 @@
+ */
+public class Parser {
+    /** Map that associates command words with their corresponding {@code Command} objects */
+    private static Logger logger = Logger.getLogger("Parser");
```
Should logger be declared `final`?

dZhei98 commented on src/main/java/seedu/main/Parser.java:
```diff
@@ -0,0 +1,165 @@
+     * @param commandString The command string entered by the user.
+     * @return The {@code Command} associated with the command string, with assigned arguments.
+     */
+    public Command parseCommand(String commandString) {
```
This function seems to be handling both parsing and argument extraction You can consider how to update this to suit SRP

DanLinhHuynh-Niwashi replied:
Created a new issue. Thank you!

dZhei98 commented on src/test/java/seedu/command/HistoryCommandTest.java:
```diff
@@ -0,0 +1,194 @@
+    private Transaction item5;
+    private Transaction item6;
+    @BeforeEach
+    public void setUp() throws Exception {
```
Good use of BeforeEach for setup

dZhei98 reviewed (commented):
Overall it seems like there are multiple functions that are quite long. You should consider refactoring these into separate files or functions to handle them instead.

isaacsaw25 commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,203 @@
+1. **commands: LinkedHashMap<String, Command>**
+  - Description: Associates command words (as keys) with their corresponding Command objects (as values).
+#### Class main methods
+1. **registerCommands(Command command): void**
```
The arguments should follow the UML style notation, namely
`command : Command` in the parameters

isaacsaw25 commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,203 @@
+    
+    ![register_command](./diagrams/parser/register-command-sequence.png)
+    
+2. **parseCommand(String commandPart): Command**
```
Arguments here as well should follow the UML notation 😄

isaacsaw25 commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,203 @@
+The `TransactionList` class is responsible for storing user transactions of different types. It also provides various
+operations that enable user to add, delete, search by (date/ category/ keywords).
+
+![TransactionList](./diagrams/TransactionList/transactionlist-class-diagram.png)
```
Great to see both Class diagrams and Sequence diagrams being used! 😸

isaacsaw25 commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,203 @@
+    
+2. **parseCommand(String commandPart): Command**
+  - **Parameters**: 
+    - `commandPart`: A string representing the command word entered by the user.
```
The formatting is a bit strange on the deployed website, where I believe you expected to place `Returns:` on a new line, but it prints on the same line as the explanation for the previous part 😰 . Perhaps you could try using a double space, which is kind of the equivalent of an escape character in Markdown. 👍

isaacsaw25 commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,203 @@
+![Category](./diagrams/category/category-class-diagram.png)
+
+### TransactionList
+The `TransactionList` class is responsible for storing user transactions of different types. It also provides various
```
In the class diagram, I feel that you could exclude the `get` functions, as their usage and implementation are quite trivial to the architecture of the class. 😅

isaacsaw25 commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,203 @@
+    - **Process**:
+        - Filters `transactions` to include only `Expense` objects with the specified `category`.
+        - Returns the filtered list of expenses.
+![SearchByCategory](./diagrams/TransactionList/transactionlist-Sequence-SearchByCategoty-diagram.png)
```
In this diagram, the 3rd self-invocation has a chain of 3 methods. Perhaps consider splitting the functions up into multiple invocations? 🐱

isaacsaw25 reviewed (commented):
Great work overall, only minor flaws in the formatting to fix 😎

NigelYeoTW commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,203 @@
+#### Class attributes
+1. **commands: LinkedHashMap<String, Command>**
+  - Description: Associates command words (as keys) with their corresponding Command objects (as values).
+#### Class main methods
+1. **registerCommands(Command command): void**
+  - **Parameters**:
+    - `command`: The `Command` object to be registered.
+  - **Process**:
+    - Retrieves the `COMMAND_WORD` field from the `Command` object
+    - Adds the word and the command to the `commands` map.
+    
```
Formatting error, also seen throughout the DG.

NigelYeoTW commented on docs/diagrams/parser/extract-arguments-sequence.png:
There should not be a class at the end of the lifeline

NigelYeoTW commented on docs/diagrams/parser/register-command-sequence.png:
self invocation should have a activation bar, comments can be placed inside notes

NigelYeoTW commented on docs/diagrams/parser/parse-command-sequence.png:
inconsistency in parameters. some have class type, some do not

NigelYeoTW commented on docs/diagrams/category/category-class-diagram.png:
additional Category class not used in diagram, Catergory2 could be named better

NigelYeoTW commented on docs/diagrams/addincomecommand/addincomecommand-class-diagram_001.png:
sequence diagram class should have ':' regardless if there is a name for the class

NigelYeoTW commented on docs/diagrams/addincomecommand/addincomecommand-class-diagram.png:
TransactionList has a missing dependency  to Transaction class

NigelYeoTW replied:
not sure if Income should have a composition to TransactionList

NigelYeoTW commented on docs/diagrams/TransactionList/transactionlist-class-diagram.png:
please follow UML notation for access modifier

NigelYeoTW commented on docs/diagrams/TransactionList/transactionlist-class-SearchKeyword-diagram.png:
inside the [] should be the terminating condition

NigelYeoTW replied:
not too sure why there were ' in the diagrams

NigelYeoTW commented on docs/DeveloperGuide.md:
```diff
@@ -1,14 +1,203 @@
+        - Aggregates and counts relevance for each match.
+        - Sorts the results by relevance and returns the list of matched transactions.
+![SearchByKeywords](./diagrams/TransactionList/transactionlist-class-SearchKeyword-diagram.png)
+5. **getExpensesByCategory(category : Category) : List<Transaction>**
```
There seems to be a formatting issue here seen throughout the DG as well

NigelYeoTW reviewed (commented):
Overall good progress, there seems to be some small issues with the diagrams but else lgtm, well done.

G13nd0n commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,240 @@
+
+### Class Main Methods
+
+1. **execute()**
```
i think you are missing the return type of the command based on the UML notation

G13nd0n replied:
Should it be `execute() : List<String>` instead?

G13nd0n commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,240 @@
+The `TransactionList` class is responsible for storing user transactions of different types. It also provides various
+operations that enable user to add, delete, search by (date/ category/ keywords).
+
+![TransactionList](./diagrams/TransactionList/transactionlist-class-diagram.png)
```
Is it possible for you to change the private and public notation to + and -?

G13nd0n commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,240 @@
 # Developer Guide
```
Good use of class diagram and sequence diagram, would it be possible for you to add the object diagram as well?

G13nd0n replied:
Good use of class and sequence diagram, would it be possible to add the object diagram as well?

G13nd0n replied:
Overall, there is formatting issue especially the arguments for a few commands do not follow the UML notation

G13nd0n commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,240 @@
+  
+  ![extract_arguments](./diagrams/parser/extract-arguments-sequence.png)
+  
+4. **splitCommandRecursively(String argumentString, String[] keywords, Map<String, String> arguments, String prevKeyword): void**
```
I think your arguments should follow the UML notation

G13nd0n replied:
The arguments should follow the UML notation, would it possible to amend it?

Commit:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-4/tp into Parser-DeveloperGuide

Commit:
Add follow-up question for expense clean up

Commit:
Disable logging

Commit:
Update test

Commit:
Fix contrustor bug

Commit:
Fix checkstyle

DanLinhHuynh-Niwashi committed:
Merge pull request #126 from DanLinhHuynh-Niwashi/DeleteCategory

Add follow-up question for expense clean up

Commit:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-4/tp into SystemTest

Commit:
Update author name

DanLinhHuynh-Niwashi committed:
Merge pull request #128 from DanLinhHuynh-Niwashi/DeleteCategory

Update author name

Commit:
Change command guide

Commit:
Update test

Commit:
Update test

Commit:
Update test

Commit:
Update test

Commit:
Update test

Commit:
Update test

Commit:
Update test

Commit:
Change naming

DanLinhHuynh-Niwashi committed:
Merge pull request #127 from DanLinhHuynh-Niwashi/SystemTest

Add system test

Gao327 committed:
Update Storage class to support save and load of the Budget and modify
other relevant classes to support this functionality

Commit:
Add some custom exception classes

DanLinhHuynh-Niwashi committed:
Merge pull request #130 from DanLinhHuynh-Niwashi/CustomExceptions

Add some custom exception classes

Gao327 committed:
Merge branch 'Branch-Gao327-07112024'

# Conflicts:
#	src/main/java/seedu/command/AddBudgetCommand.java
#	src/main/java/seedu/transaction/Transaction.java

Gao327 committed:
Update EXPECTED.TXT

DanLinhHuynh-Niwashi committed:
Update UserGuide.md

Gao327 committed:
Merge pull request #129 from Gao327/Branch-Gao327-07112024

Update Storage class to support save and load of the Budget

DanLinhHuynh-Niwashi committed:
Merge pull request #131 from DanLinhHuynh-Niwashi/UserGuide-1

Update UserGuide.md

DanLinhHuynh-Niwashi committed:
Merge pull request #1 from AY2425S1-CS2113-W10-4/master

Merge master

Gao327 committed:
Update main class to include assertion for BudgetTracker

YukeeHong committed:
add test for AddExpenseCommand, UpdateCategoryCommand, KeywordsSearchCommand, DeleteTransactionCommand, AddCategoryCommand, LocalDateTimeAdapter

Gao327 committed:
Update gitignore

Gao327 committed:
Merge pull request #132 from Gao327/Branch-Gao327-07112024

Update main class to include assertion for BudgetTracker

DanLinhHuynh-Niwashi committed:
Update DeveloperGuide.md

DanLinhHuynh-Niwashi committed:
Merge pull request #134 from DanLinhHuynh-Niwashi/Parser-DeveloperGuide

Reformat developer guide

YukeeHong committed:
add test for AddExpenseCommand, UpdateCategoryCommand, KeywordsSearchCommand, DeleteTransactionCommand, AddCategoryCommand, LocalDateTimeAdapter

YukeeHong committed:
checkstyle

DanLinhHuynh-Niwashi committed:
Fix format error

YukeeHong committed:
checkstyle

YukeeHong committed:
checkstyle

DanLinhHuynh-Niwashi committed:
Add base Command class guide

DanLinhHuynh-Niwashi committed:
Update DeveloperGuide.md

DanLinhHuynh-Niwashi committed:
Merge pull request #135 from DanLinhHuynh-Niwashi/Parser-DeveloperGuide

Add Command developer guide

DanLinhHuynh-Niwashi committed:
Add table of contents

DanLinhHuynh-Niwashi committed:
Merge pull request #136 from DanLinhHuynh-Niwashi/Parser-DeveloperGuide

Add table of contents

YukeeHong committed:
test.sh

YukeeHong committed:
Merge branch 'master' into add_test

YukeeHong committed:
test.sh update

DanLinhHuynh-Niwashi committed:
Add glossary and test guide

DanLinhHuynh-Niwashi committed:
Merge pull request #137 from DanLinhHuynh-Niwashi/Parser-DeveloperGuide

Add glossary and test guide

DanLinhHuynh-Niwashi committed:
Merge pull request #133 from YukeeHong/add_test

Add tests with CI error

Commit:
Revert "checkstyle"

This reverts commit c2f7a21cd751710515788b57a816f51955853cb8.

Commit:
Update test file

Gao327 committed:
Merge pull request #138 from DanLinhHuynh-Niwashi/SystemTest-1

Change expected.txt

Commit:
Add new static messages

Commit:
Update test date to current for future test run

Commit:
Remove Duke

Commit:
Update test with budget data file

Commit:
Update gitignore to untrack future data file

Commit:
Update checkstyle

Commit:
Update test

DanLinhHuynh-Niwashi committed:
Merge pull request #139 from DanLinhHuynh-Niwashi/SystemTest-1

System test

YukeeHong committed:
add unused code

YukeeHong committed:
Merge pull request #140 from YukeeHong/add_unused_code

add unused code

DanLinhHuynh-Niwashi committed:
Update Command Developer guide

DanLinhHuynh-Niwashi committed:
Fix format error

Commit:
Update datetime error message

YukeeHong committed:
add user guide for track command

Commit:
Update PPP header

YukeeHong committed:
Merge pull request #141 from YukeeHong/update_userguide

add user guide for track command

Commit:
Reformat the PPP

DanLinhHuynh-Niwashi committed:
Merge pull request #142 from DanLinhHuynh-Niwashi/Parser-DeveloperGuide

Parser developer guide

DanLinhHuynh-Niwashi committed:
Merge pull request #143 from DanLinhHuynh-Niwashi/Documentation

Documentation

Gao327 committed:
Create StorageTest JUnit Test

Gao327 committed:
Fix Code Style

Commit:
Add date parsing rules

DanLinhHuynh-Niwashi committed:
Merge pull request #145 from DanLinhHuynh-Niwashi/ViewHistory

Add date parsing rules

Gao327 committed:
Merge pull request #144 from Gao327/Branch-Gao327-07112024

Create StorageTest JUnit Test

Gao327 committed:
Add directory to store unused code

Gao327 committed:
Remove LocalDateTimeAdapterTest from src

Oshen27 committed:
Update UserGuide to include AddBudgetCommand

Gao327 committed:
Merge branch 'Branch-Gao327-07112024'

Gao327 committed:
Merge branch 'AY2425S1-CS2113-W10-4:master' into master

Oshen27 committed:
Merge commit '3d3f97483d511af3315d49b97ea719470dc54dd8' into UserGuide

Gao327 committed:
Merge pull request #146 from Gao327/Branch-Gao327-07112024

Add unused code directory

Oshen27 committed:
Merge pull request #147 from Oshen27/UserGuide

Update User guide

Gao327 committed:
Merge branch 'AY2425S1-CS2113-W10-4:master' into master

Commit:
Add new diagram for Parser

DanLinhHuynh-Niwashi committed:
Merge pull request #148 from DanLinhHuynh-Niwashi/Documentation

Add new diagram for Parser

Gao327 committed:
Update Main class to show monthly income and expense on UI

Gao327 committed:
Update EXPECTED.TXT

Gao327 committed:
Merge pull request #149 from Gao327/Branch-Gao327-07112024

Update Main class to show monthly income and expense on UI

Gao327 committed:
Merge branch 'master' of https://github.com/Gao327/tp

DanLinhHuynh-Niwashi committed:
Merge pull request #2 from AY2425S1-CS2113-W10-4/master

Pull from master

DanLinhHuynh-Niwashi committed:
Update UserGuide.md

DanLinhHuynh-Niwashi committed:
Merge pull request #150 from DanLinhHuynh-Niwashi/Documentation

Documentation

DanLinhHuynh-Niwashi committed:
Update UserGuide.md format

DanLinhHuynh-Niwashi committed:
Merge pull request #151 from DanLinhHuynh-Niwashi/Documentation

Update UserGuide.md format

DanLinhHuynh-Niwashi committed:
Update UserGuide.md

DanLinhHuynh-Niwashi committed:
Merge pull request #152 from DanLinhHuynh-Niwashi/Documentation

Update UserGuide.md

Oshen27 committed:
Update AddIncomeCommand Class Diagram

Oshen27 committed:
Merge pull request #153 from Oshen27/DeveloperGuide

Update AddIncomeCommand Class Diagram

Commit:
Update search command for partial match

DanLinhHuynh-Niwashi committed:
Merge pull request #154 from DanLinhHuynh-Niwashi/SearchCommand

Update search command for partial match

DanLinhHuynh-Niwashi committed:
Update UserGuide.md links

DanLinhHuynh-Niwashi committed:
Update DeveloperGuide.md update links

DanLinhHuynh-Niwashi committed:
Merge pull request #155 from DanLinhHuynh-Niwashi/Documentation

Documentation

DanLinhHuynh-Niwashi committed:
Update UserGuide.md

DanLinhHuynh-Niwashi committed:
Merge pull request #156 from DanLinhHuynh-Niwashi/Documentation

Update UserGuide.md

Commit:
Add help guide

Commit:
Update test

DanLinhHuynh-Niwashi committed:
Merge pull request #157 from DanLinhHuynh-Niwashi/SearchCommand

Add help guide

Commit:
Add start date - end date validation

Commit:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-4/tp into ViewHistory

Commit:
Resolve confusing indexing of transaction list

Commit:
Handle ignore case category

Commit:
Fix checkstyle violation

DanLinhHuynh-Niwashi committed:
Merge pull request #203 from DanLinhHuynh-Niwashi/ViewHistory

Handle TransactionList confusing bugs and inconsistent category comparation

DanLinhHuynh-Niwashi committed:
Correct link for PPP

DanLinhHuynh-Niwashi committed:
Update danlinhhuynh-niwashi.md

DanLinhHuynh-Niwashi committed:
Merge pull request #204 from DanLinhHuynh-Niwashi/Documentation

Update PPP

DanLinhHuynh-Niwashi committed:
Shorten the PPP

Commit:
Show transaction list after each change

DanLinhHuynh-Niwashi committed:
Merge pull request #205 from DanLinhHuynh-Niwashi/ViewHistory

Show transaction list after each change

Commit:
Update diagrams

DanLinhHuynh-Niwashi committed:
Update diagram for Parser

DanLinhHuynh-Niwashi committed:
Update Parser guide

DanLinhHuynh-Niwashi committed:
Merge pull request #206 from DanLinhHuynh-Niwashi/Documentation

Update PPP and fix Parser bugs in DG

DanLinhHuynh-Niwashi committed:
Update danlinhhuynh-niwashi.md

DanLinhHuynh-Niwashi committed:
Merge pull request #207 from DanLinhHuynh-Niwashi/Documentation

Update danlinhhuynh-niwashi.md

Gao327 committed:
Merge branch 'master' of https://github.com/Gao327/tp

Gao327 committed:
Include upper limit for amount parsing

Gao327 committed:
Update upper limit for amount per transaction

Gao327 committed:
Update EXPECTED.TXT

Gao327 committed:
Merge pull request #209 from Gao327/Branch-Gao327-11112024

Upper Limit per transaction

Gao327 committed:
Create DescriptionUtil class to facilitate description parsing in
transactions and categories

Gao327 committed:
Update the upper limit of argument length to 120 characters

Gao327 committed:
Modify error message printing for description format error

Gao327 committed:
Update description length limit to 40 characters

Gao327 committed:
Merge pull request #210 from Gao327/Branch-Gao327-11112024

Create upper limit for description length

Gao327 committed:
Modify save and load functionality to handle altered files

Gao327 committed:
Merge pull request #211 from Gao327/Branch-Gao327-11112024

Modify save and load

Gao327 committed:
Correct transactionlist-class-diagram

Gao327 committed:
Merge branch 'master' into Branch-Gao327-11112024

Gao327 committed:
Merge pull request #212 from Gao327/Branch-Gao327-11112024

Correct DG diagram

Gao327 committed:
Update code for addincomecommand-class-diagram

DanLinhHuynh-Niwashi committed:
Update UserGuide.md

Update add-expense in response to: #158, #195, #172, #169, #158
Add note about order in response to: #189
Add note about empty command in response to: #171
Delete the 'clear' command in response to: #179
Edit command summary in response to: #162, #159
Add 'bye'  command in response to: #161
Fix indentation in response to: #160

DanLinhHuynh-Niwashi committed:
Merge pull request #213 from DanLinhHuynh-Niwashi/Documentation

Update UserGuide.md for syntax issue and add notes

Gao327 committed:
modify getexpensebycategory-sequence-diagram

Gao327 committed:
Merge pull request #214 from Gao327/Branch-Gao327-11112024

Update DG diagram

Gao327 committed:
Update acknowledgment in DeveloperGuide

Gao327 committed:
Merge pull request #215 from Gao327/Branch-Gao327-11112024

Update acknowledgment in DeveloperGuide

Gao327 committed:
Update DG diagram

Gao327 committed:
Merge pull request #216 from Gao327/Branch-Gao327-11112024

Update DG diagram

Oshen27 committed:
Add error handling for add category

Oshen27 committed:
Merge commit '73d1f13f4c3c90384f66d6c20c6afc7b00598c5e'

# Conflicts:
#	src/main/java/seedu/command/AddCategoryCommand.java

YukeeHong committed:
update DeveloperGuide.md

YukeeHong committed:
update DeveloperGuide.md

YukeeHong committed:
update DeveloperGuide.md

YukeeHong committed:
Merge pull request #217 from YukeeHong/master

update user guide for category

YukeeHong committed:
update DeveloperGuide.md

YukeeHong committed:
Merge pull request #218 from YukeeHong/master

update DeveloperGuide.md

YukeeHong committed:
update UserGuide.md

YukeeHong committed:
Merge pull request #219 from YukeeHong/update_ug

update UserGuide.md

Oshen27 committed:
Improve error messages

Commit:
Add new exception classes

Commit:
Update storage to handle invalid input

Commit:
Add new exception handler

YukeeHong committed:
update ppp

YukeeHong committed:
Merge pull request #220 from YukeeHong/update_ppp

update ppp

Oshen27 committed:
Update naming to following convention and improve error message

Commit:
Update command format

Commit:
Update follow up question handling and test

Oshen27 committed:
Add view all budgets functionality

Oshen27 committed:
Add delete budget command

Oshen27 committed:
Update Tests

Oshen27 committed:
Fix checkstyle and tests

Oshen27 committed:
Remove unused imports

Oshen27 committed:
Fix test

Oshen27 committed:
Revert test

Commit:
Handle future transaction error

Commit:
Update datetimeutil to parse to 0000

Commit:
Handle unsaved error

DanLinhHuynh-Niwashi committed:
Merge pull request #221 from Oshen27/BudgetTracker

Update Budget related code

Oshen27 committed:
Update UG with delete budget and view budget

Commit:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-4/tp into UpdateErrorHandling

# Conflicts:
#	src/main/java/seedu/utils/AmountUtils.java
#	src/test/java/seedu/command/AddExpenseCommandTest.java
#	src/test/java/seedu/utils/AmountUtilsTest.java

Commit:
Edit return message

Commit:
Add final save

Commit:
Fix checkstyle

Commit:
Add message on lack arguments

DanLinhHuynh-Niwashi committed:
Add manual test guide

DanLinhHuynh-Niwashi committed:
Merge pull request #222 from DanLinhHuynh-Niwashi/UpdateErrorHandling

Update error handling

DanLinhHuynh-Niwashi committed:
Merge pull request #223 from DanLinhHuynh-Niwashi/Documentation

Add manual test guide

Commit:
Add view history diagram

Oshen27 committed:
Merge commit 'a4ef32ffa09bf47b1ab740c69f25f76d5cb7ef61' into UserGuide

Commit:
Add diagrams

Commit:
Fix diagram

DanLinhHuynh-Niwashi committed:
Add income, command class

DanLinhHuynh-Niwashi committed:
Update DeveloperGuide.md format

Commit:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W10-4/tp into Documentation

# Conflicts:
#	docs/diagrams/category/category-class-diagram.png
#	docs/plantUMLCode/category-class-diagram.puml

Commit:
Merge branch 'Documentation' of https://github.com/DanLinhHuynh-Niwashi/tp into Documentation

# Conflicts:
#	docs/DeveloperGuide.md

DanLinhHuynh-Niwashi committed:
Update danlinhhuynh-niwashi.md

DanLinhHuynh-Niwashi committed:
Merge pull request #224 from DanLinhHuynh-Niwashi/Documentation

Add documentation for ViewHistory and Transaction

Commit:
Fix style

Commit:
Merge branch 'Documentation' of https://github.com/DanLinhHuynh-Niwashi/tp into Documentation

Commit:
Fix links

Commit:
Reformat Developer guide

DanLinhHuynh-Niwashi committed:
Merge pull request #225 from DanLinhHuynh-Niwashi/Documentation

Documentation

Commit:
Fix indentation of DG

Commit:
Update diagram

Commit:
Fix PPP format

DanLinhHuynh-Niwashi committed:
Merge pull request #226 from DanLinhHuynh-Niwashi/Documentation

Fix indentation of DG

Commit:
Add high-level architecture

Commit:
Add architecture documentation

DanLinhHuynh-Niwashi committed:
Merge pull request #227 from DanLinhHuynh-Niwashi/Documentation

Documentation for Architecture

Oshen27 committed:
Merge commit '9bca60af0df0331bb45316da7d456ec27a891eea' into UserGuide

YukeeHong committed:
update ppp

YukeeHong committed:
update ppp

YukeeHong committed:
update ppp

DanLinhHuynh-Niwashi committed:
Update UserGuide.md add FAQ and notes

Oshen27 committed:
Update UG with example output

DanLinhHuynh-Niwashi committed:
Merge pull request #229 from DanLinhHuynh-Niwashi/Documentation

Update UserGuide.md add FAQ and notes

YukeeHong committed:
Merge pull request #230 from YukeeHong/update_ppp

Update ppp

YukeeHong committed:
update ppp

Oshen27 committed:
Merge commit 'bd53e6efede1af45056f7cb480cf8cc5b339c7b2' into Oshen27-UserGuide

# Conflicts:
#	docs/UserGuide.md

Gao327 committed:
Modify UserGuide

Gao327 committed:
Merge pull request #232 from Gao327/Branch-Gao327-final

Modify UserGuide

Oshen27 committed:
Update PPP

DanLinhHuynh-Niwashi committed:
Merge branch 'master' into Oshen27-UserGuide

DanLinhHuynh-Niwashi committed:
Merge pull request #233 from Oshen27/Oshen27-UserGuide

Update User guide

DanLinhHuynh-Niwashi committed:
Update UserGuide.md change formatting

DanLinhHuynh-Niwashi committed:
Merge pull request #234 from DanLinhHuynh-Niwashi/Documentation

Update UserGuide.md change formatting

YukeeHong committed:
update ppp

YukeeHong committed:
Merge pull request #235 from YukeeHong/update_ppp

Update ppp

Commit:
Change diagram and PPP

DanLinhHuynh-Niwashi committed:
Merge pull request #236 from DanLinhHuynh-Niwashi/Documentation

Change diagram and PPP

Gao327 committed:
Update DG diagram

Commit:
Update user guide for pdf print

DanLinhHuynh-Niwashi committed:
Merge pull request #237 from Gao327/Branch-Gao327-final

Update DG diagram

DanLinhHuynh-Niwashi committed:
Merge pull request #238 from DanLinhHuynh-Niwashi/Documentation

Update user guide for pdf print

Commit:
Reformat UG

DanLinhHuynh-Niwashi committed:
Merge pull request #239 from DanLinhHuynh-Niwashi/Documentation

Reformat UG

Commit:
Reformat for printing

DanLinhHuynh-Niwashi committed:
Merge pull request #240 from DanLinhHuynh-Niwashi/Documentation

Reformat for printing

Commit:
Fix link

DanLinhHuynh-Niwashi committed:
Merge pull request #241 from DanLinhHuynh-Niwashi/Documentation

Fix link

Commit:
Fix link

DanLinhHuynh-Niwashi committed:
Merge pull request #242 from DanLinhHuynh-Niwashi/Documentation

Fix link

Gao327 committed:
update

Gao327 committed:
Merge pull request #243 from Gao327/master

update
